vector/src/main/java/im/vector/app/features/crypto/quads/SharedSecureStorageViewModel.kt

```diff
@@ -167,10 +166,14 @@ class SharedSecureStorageViewModel @AssistedInject constructor(
         if (state.checkingSSSSAction is Loading) return@withState // ignore
         when (state.step) {
             SharedSecureStorageViewState.Step.EnterKey -> {
-                setState {
-                    copy(
-                            step = SharedSecureStorageViewState.Step.EnterPassphrase
-                    )
+                if (state.hasPassphrase) {
```

Comment:
this block is the fix, everything else in this PR is for the unit tests

Reply:
perfect, thanks!